src/backend/bs_dispatch

```diff
@@ -72,18 +102,12 @@ if ($BSConfig::secure_sandboxes) {
   $secure_sandboxes{$_} = 1 for qw{xen kvm zvm};
 }
 
-my $testmode;
-while (@ARGV) {
-  if ($ARGV[0] eq '--test-mode' || $ARGV[0] eq '--testmode') {
-    shift @ARGV;
-    $testmode = 1;
-  } elsif ($ARGV[0] eq '--test-constraints') {
-    shift @ARGV;
-    my $package = shift @ARGV;
-    my $architecture = shift @ARGV;
-    my $workerinfo_file = shift @ARGV;
-    my $constraints_file = shift @ARGV;
-    my $constraintsprj_file = shift @ARGV;
+# copy @ARGV to keep it untouched in case of restart
+my $options = parse_options(@ARGV);
+
+if ($options->{'testconstraints'}) {
+    my ($package,$architecture,$workerinfo_file,$constraints_file,$constraintsprj_file) = split(',',$options->{'testconstraints'});
+
```

Comment:
Why that split? Just make testconstraints a bool and use the remaining args.

Reply:
okay, done